### PR TITLE
sof-test: rename OPT_PARM_lst and OPT_VALUE_lst

### PR DESCRIPTION
**Pull request by aiChaoSONG**
This PR renames two variable names used to define script command line options, OPT_PARM_lst to OPT_HAS_ARG, OPT_VALUE_lst to OPT_ARG

No functional change, only repo wide search and replace. check discussion https://github.com/thesofproject/sof-test/issues

### CODE REVIEW

File: case-lib/opt.sh [hunk 1]
```diff
@@ -3,28 +3,28 @@
...
 # option setup && parse function
 func_opt_parse_option()
 {
     # OPT_NAME     (long) option name
     # OPT_DESC    short sentence describing the option
-    # OPT_PARM_lst    0 or 1: number of argument required
-    # OPT_VALUE_lst   default value overwritten by command line
+    # OPT_HAS_ARG    0 or 1: number of argument required
+    # OPT_ARG   default value overwritten by command line
     #                 input if any. Set to 0 or 1 when PARM=0
```
Comment (xiulipan) [inline case-lib/opt.sh:15]:
```suggestion
    #                 input if any. Set to 0 or 1 when OPT_HAS_ARG    =0
```

And try to refine the text here to make it more easy to understand

File: case-lib/opt.sh [hunk 2]
```diff
@@ -3,28 +3,28 @@
...
 
     # for example
     # OPT_NAME['r']='remote'
     # OPT_DESC['r']='Run for the remote machine'
-    # OPT_PARM_lst['r']=1
-    # OPT_VALUE_lst['r']='' ## if PARM=1, must be set, if PARM=0, can ignore
+    # OPT_HAS_ARG['r']=1
+    # OPT_ARG['r']='' ## if PARM=1, must be set, if PARM=0, can ignore
```
Comment (xiulipan) [inline case-lib/opt.sh:21]: PARM in the comments also need to be changed

File: case-lib/opt.sh [hunk 3]
```diff
@@ -82,19 +82,19 @@ func_opt_parse_option()
                 # display short option
                 [ "$i" ] && printf '    -%s' "$i"
                 # have parameter
```
Comment (xiulipan) [inline case-lib/opt.sh:84]:
```suggestion
                # have argument
```

**Review by xiulipan — CHANGES_REQUESTED** [review]
Please also go through the inline comments to see if we need any update.

File: case-lib/opt.sh (hunk 1 above)
Comment (aiChaoSONG) [reply case-lib/opt.sh]: Updated

**Comment by marc-hb** [comment]
As I wrote in #572, `OPT_ARG` is misleading because it also stores the boolean state for options that have no arg.

File: case-lib/opt.sh [hunk 4]
```diff
@@ -3,28 +3,28 @@
...
+    # OPT_ARG   default value overwritten by command line
+    #                 input if any. Set to 0 or 1 when OPT_HAS_ARG=0
 
     # for example
     # OPT_NAME['r']='remote'
     # OPT_DESC['r']='Run for the remote machine'
-    # OPT_PARM_lst['r']=1
-    # OPT_VALUE_lst['r']='' ## if PARM=1, must be set, if PARM=0, can ignore
+    # OPT_HAS_ARG['r']=1
+    # OPT_ARG['r']='' ## if OPT_HAS_ARG=1, must be set, if OPT_HAS_ARG=0, can ignore
```
Comment (marc-hb) [inline case-lib/opt.sh:21]: This comment is very misleading (I think it did misled you)

**Comment by aiChaoSONG** [comment]
> As I wrote in #572, `OPT_ARG` is misleading because it also stores the boolean state for options that have no arg.

but OPT_HAS_ARG has already done this for us, I mean if OPT_HAS_ARG is 0, we can leave OPT_ARG unspecified. this requires some changes in opt.sh(can be done later).  for option require extra arg, we do
```
OPT_NAME['o']='option_name'
OPT_HAS_ARG['a']=1
OPT_ARG['a']=some_name
OPT_DESC['a']='some description'
```
and for option doesn't require extra arg, we do
```
OPT_NAME['o']='option_name'
OPT_HAS_ARG['a']=0
OPT_DESC['a']='some description'
```

File: case-lib/opt.sh (hunk 4 above)
Comment (aiChaoSONG) [reply case-lib/opt.sh:21]: Yes, I think I will delete the demo comment here sometime later. and as this PR require rebase, I will drop the comment fix commit. The clean up of opt.sh is just startup.

**Comment by marc-hb** [comment]
> but OPT_HAS_ARG has already done this for us, I mean if OPT_HAS_ARG is 0, we can leave OPT_ARG unspecified. 

Then how do you know if the flag is used?

**Comment by aiChaoSONG** [comment]
> Then how do you know whether the flag has been used?

This is what explained in opt.sh by Bin

**Review by marc-hb — CHANGES_REQUESTED** [review]
> This is what explained in opt.sh by Bin

The code is from Bin but the explanation / comment is from me (check the git blame): options without arg use OPT_VALUE_lst as a boolean, that's where the information is stored and AFAIK nowhere else. This code is not the easiest to understand (this comment cost me a lot of time) but it's very well tested and it works so let's not bother changing it unless we really need to.

This PR and issue started as a large but very simple and mechanical rename easy to review but now we're mixing that up with a code change tangent.

**Comment by aiChaoSONG** [comment]
> This PR and issue started as a large but very simple and mechanical rename easy to review but now we're mixing that up with a code change tangent.

Ok, Let's use OPT_VAL currently.

**Review by xiulipan — APPROVED** [review]
LGTM, let's only do name change first. Local git grep checked.

**Comment by aiChaoSONG** [comment]
The failure on TGLH_RVP_HDA is not related to this PR. and reviews are all resolved.

**Pull request merged**